src/main/java/io/appium/java_client/android/AndroidMobileOptions.java

```diff
@@ -0,0 +1,1716 @@
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.appium.java_client.android;
+
+import io.appium.java_client.remote.AndroidMobileCapabilityType;
+import io.appium.java_client.remote.MobileOptions;
+import io.appium.java_client.remote.MobilePlatform;
+import org.openqa.selenium.Capabilities;
+import org.openqa.selenium.chrome.ChromeOptions;
+
+import java.time.Duration;
+import java.util.List;
+
+public class AndroidMobileOptions extends MobileOptions<AndroidMobileOptions> {
+
+    /**
+     * Creates new instance with platformName specified for Android.
+     */
+    public AndroidMobileOptions() {
+        setPlatformName(MobilePlatform.ANDROID);
+    }
+
+    /**
+     * Creates new instance with provided capabilities capabilities.
+     *
+     * @param source is Capabilities instance to merge into new instance
+     */
+    public AndroidMobileOptions(Capabilities source) {
+        this();
+        merge(source);
+    }
+
+    /**
+     * Set the Timeout in milliseconds used to wait for adb command execution.
+     *
+     * @param ms is the number of milliseconds to wait for adb command execution.
+     * @return this MobileOptions, for chaining.
+     * @see AndroidMobileCapabilityType#ADB_EXEC_TIMEOUT
+     */
+    public AndroidMobileOptions setAdbExecTimeout(Duration ms) {
+        return amend(AndroidMobileCapabilityType.ADB_EXEC_TIMEOUT, ms);
+    }
+
+    /**
+     * Get the Timeout in milliseconds used to wait for adb command execution.
+     *
+     * @return Duration to wait for adb command execution.
+     * @see AndroidMobileCapabilityType#ADB_EXEC_TIMEOUT
+     */
+    public Duration getAdbExecTimeout() {
+        return (Duration) getCapability(AndroidMobileCapabilityType.ADB_EXEC_TIMEOUT);
+    }
+
+    /**
+     * Set the port used to connect to the ADB server.
+     *
+     * @param port is the port to connect to the ADB server.
+     * @return this MobileOptions, for chaining.
+     * @see AndroidMobileCapabilityType#ADB_PORT
+     */
+    public AndroidMobileOptions setAdbPort(String port) {
+        return amend(AndroidMobileCapabilityType.ADB_PORT, port);
+    }
+
+    /**
+     * Get the port used to connect to the ADB server.
+     *
+     * @return String of the port used to connect to the ADB server.
+     * @see AndroidMobileCapabilityType#ADB_PORT
+     */
+    public String getAdbPort() {
+        return (String) getCapability(AndroidMobileCapabilityType.ADB_PORT);
```

Comment:
I think we should handle both numbers and strings here properly.

Reply:
I'm not sure what you mean. If we set it as an Integer, it needs to be returned as an Integer.
Updated: c5fbb3dc67a0891a002bbca235595fe13a8c3351

Reply:
integer is fine 👍